TOOLS/lua/autoload.lua

```diff
@@ -94,35 +94,17 @@ table.filter = function(t, iter)
     end
 end
 
--- splitbynum and alnumcomp from alphanum.lua (C) Andre Bogus
--- Released under the MIT License
--- http://www.davekoelle.com/files/alphanum.lua
-
--- split a string into a table of number and string values
-function splitbynum(s)
-    local result = {}
-    for x, y in (s or ""):gmatch("(%d*)(%D*)") do
-        if x ~= "" then table.insert(result, tonumber(x)) end
-        if y ~= "" then table.insert(result, y) end
-    end
-    return result
-end
-
-function clean_key(k)
-    k = (' '..k..' '):gsub("%s+", " "):sub(2, -2):lower()
-    return splitbynum(k)
-end
+-- alphanum sorting for humans in Lua
+-- http://notebook.kulchenko.com/algorithms/alphanumeric-natural-sorting-for-humans-in-lua
 
 -- compare two strings
-function alnumcomp(x, y)
-    local xt, yt = clean_key(x), clean_key(y)
-    for i = 1, math.min(#xt, #yt) do
-        local xe, ye = xt[i], yt[i]
-        if type(xe) == "string" then ye = tostring(ye)
-        elseif type(ye) == "string" then xe = tostring(xe) end
-        if xe ~= ye then return xe < ye end
-    end
-    return #xt < #yt
+function alphanumsort(a, b)
+  local function padnum(d)
+    local dec, n = string.match(d, "(%.?)0*(.+)")
+    return #dec > 0 and ("%.12f"):format(d) or ("%s%03d%s"):format(dec, #n, n)
```

Comment:
Is `dec` in the `:format()` even doing anything?
It only gets there when `dec` is an empty string, and so formatting it will always result in nothing.

Reply:
You mean change it to `return #dec > 0 and ("%.12f"):format(d) or ("%03d%s"):format(#n, n)`?
It didn't bring any changes in my test, but I want to keep it in case of some unknown effect. After all, it is insignificant.

Reply:
FYI, I've added it without the `dec` in https://github.com/tomasklaen/uosc/pull/338. Let's see if anyone notices any problems, but I very much doubt it.